src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -2534,19 +2534,33 @@ bool RTPSParticipantImpl::is_reader_ignored(
 bool RTPSParticipantImpl::ignore_participant(
         const GuidPrefix_t& participant_guid)
 {
-    if (!is_participant_ignored(participant_guid))
+    if (participant_guid == m_guid.guidPrefix)
     {
+        EPROSIMA_LOG_WARNING(RTPS_PARTICIPANT, "A participant is unable to ignore itself");
+        return false;
+    }
+    {
+        shared_lock<eprosima::shared_mutex> _(mp_builtinProtocols->getDiscoveryMutex());
+
+        for (auto server_it = m_att.builtin.discovery_config.m_DiscoveryServers.begin();
+                server_it != m_att.builtin.discovery_config.m_DiscoveryServers.end(); server_it++)
         {
+            if (server_it->guidPrefix == participant_guid)
             {
-                std::unique_lock<shared_mutex> _(ignored_mtx_);
-                ignored_participants_.insert(participant_guid);
+                EPROSIMA_LOG_WARNING(RTPS_PARTICIPANT, "Cannot ignore one of this participant Discovery Servers");
+                return false;
             }
-            pdp()->remove_remote_participant(GUID_t(participant_guid, c_EntityId_RTPSParticipant),
-                    ParticipantDiscoveryInfo::DISCOVERY_STATUS::IGNORED_PARTICIPANT);
         }
-        return true;
     }
-    return false;
+    {
+        std::unique_lock<shared_mutex> _(ignored_mtx_);
+        ignored_participants_.insert(participant_guid);
+    }
+    pdp()->remove_remote_participant(GUID_t(participant_guid, c_EntityId_RTPSParticipant),
+            ParticipantDiscoveryInfo::DISCOVERY_STATUS::IGNORED_PARTICIPANT);
```

Comment:
Shouldn't we do this only if the participant is not already ignored?

Reply:
Adding it to the set is idempotent, and `remove_remote_participant` will do nothing if the participant has already been removed from the PDP database